utils/src/arg_parser.rs

```diff
@@ -0,0 +1,582 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::borrow::Borrow;
+use std::collections::HashSet;
+use std::env;
+use std::fmt;
+use std::hash::{Hash, Hasher};
+use std::process;
+use std::result;
+
+pub type Result<T> = result::Result<T, Error>;
+
+const DOUBLE_DASH: &str = "--";
+const HELP_ARG: &str = "--help";
+const FLAG_PROVIDED: &str = "true";
+
+/// Errors associated with parsing and validating arguments.
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// The required argument was not provided.
+    MissingArgument(String),
+    /// A value for the argument was not provided.
+    MissingValue(String),
+    /// The provided argument was not expected.
+    UnexpectedArgument(String),
+    /// The argument was provided more than once.
+    DuplicatedArgument(String),
```

Comment:
nit: IMO, this should be `DuplicateArgument(String)` (i.e. the arg is a duplicate).